extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java

```diff
@@ -155,8 +165,20 @@ public QuartzScheduler(SchedulerContext context, QuartzSupport quartzSupport, Co
                                 scheduled);
                     }
                 }
-            } catch (SchedulerException e) {
+                if (manageTx) {
+                    handle.get().commit();
+                }
+            } catch (Throwable e) {
+                if (manageTx) {
+                    try {
+                        handle.get().rollback();
+                    } catch (SystemException ex) {
+                        LOGGER.error("Unable to rollback transaction", ex);
+                    }
+                }
                 throw new IllegalStateException("Unable to create Scheduler", e);
+            } finally {
+                handle.close();
```

Comment:
`InstanceHandle` is `AutoCloseable` so you can just use try-with-catch (although it's not necessary in this case because the `UserTransaction` is `@ApplicationScoped`, but better safe than sorry ;-).